### DISCUSSION TITLE
Switching over from Sequel Pro

### DISCUSSION THREAD

**Discussion by bhjohns** [Sequel-Ace/Sequel-Ace#749] [Q&A]
I just set up on a new computer and thought it was a good time to switch from Sequel Pro to Sequel Ace.  Unfortunately, after installing and getting all my favorites imported, none of my connections work in Sequel Ace.  My connections are simple TCP/IP connections.  I have tested and it still works in Sequel Pro and even from the command line like so:
`mysql --host=***** --user=***** --password=***** myDatabase`
but when I try the same information in Sequel Ace I get 

> Connection Failed
> Unable to connect to host podsjqe-mysql.marketo.org, or the request timed out.
> Be sure that the address is correct and that you have the necessary privileges, or try increasing the connection timeout (currently 10 seconds).
> MySQL said: Authentication failed.
> Authentication failed.

Am I missing something?  Why would it work in Sequel Pro and even command line but fail in sequel ace?

Any help troubleshooting is welcome and appreciated, otherwise I'll keep trying to use Sequel Pro I suppose.

**Comment by Jason-Morcos** [comment]
Odd
Are all your authentications simple username/password authentications? One thing that's likely is your SSH config has something special. In Sequel Pro (and in the terminal) your SSH config file is used to initialize SSH connections. Sequel Ace is sandboxed so it can't access your SSH config unless you manually allow it to do so. I would try choosing your ssh config file, and making sure Sequel Ace has access to any private key files it should have access to!

https://sequel-ace.com/get-started/
see "Sequel Ace doesn’t read my ~/.ssh/config parameters."

**Reply by bhjohns** [reply]
@Jason-Morcos  thanks for the response.  Yes all the authentications are simple username/password.  None of these connections are SSH just IP/TCP so the SSH config shouldn't matter should it?  I tried messing with the connection through SSH and selected my SSH config and key, but still had trouble.  In general, the database is accessible directly via IP with username and password.  The only other restriction is having to be on a VPN, but as long as the VPN is on then I don't see why that would affect things.

So essentially once the VPN in turned on, then I can access the database from terminal with host, username, password.  I can do the same with Sequel Pro with the "Standard" Connection (IP/TCP) only filling in the host, username, and password.  But, when I do the exact same steps with Sequel Ace I get Connection Failed.  The error happens immediately, so it shouldn't be timing out.

**Reply by Jason-Morcos** [reply]
Hmmm I don't know much about connecting through a vpn. @Sequel-Ace/all perhaps someone else knows a little?

**Reply by bhjohns** [reply]
Doing some additional testing I don't think the problem is related to the VPN as I have found 1 database that I am successfully able to connect to, but still am unable to connect to the rest.  The database that connects is more of a test clone and not one of the regular databases so perhaps there's something setup in the databases that is rejecting the attempt to authenticate?  I'm not an admin so am unable to tweak database settings, but it's still weird to me that it only rejects the attempt from Sequel Ace and not from Terminal or Sequel Pro.

**Comment by stychos** [comment]
I used to connect via VPN, and it works fine:
<img width="493" alt="Screenshot 2021-01-06 at 22 25 59" src="https://user-images.githubusercontent.com/1531639/103817240-f1e6ee00-506e-11eb-98b7-a92ed086404f.png">

@bhjohns can you check if you don't have hardcoded ip-hostname pairs in your `/etc/hosts` for that VPN hosts? Your configured DNS on the VPN connection must answer with proper addresses for that hostnames. For example, my VPN has a configured DNS `192.168.0.254` which is also in the VPN and which answers with `192.168.0.1` for that `hydra` hostname.

**Reply by bhjohns** [reply]
If there was a problem with the host mapping I don't see how it would work in Sequel Pro and Terminal, but for me both of those work fine and just not from Sequel Ace.

Doing some additional testing I don't think the problem is related to the VPN as I have found 1 database that I am successfully able to connect to, but still am unable to connect to the rest.  The database that connects is more of a test clone and not one of the regular databases so perhaps there's something setup in the databases that is rejecting the attempt to authenticate?  I'm not an admin so am unable to tweak database settings, but it's still weird to me that it only rejects the attempt from Sequel Ace and not from Terminal or Sequel Pro.

**Reply by Jason-Morcos** [reply]
What version of MySQL are we talking about? Ace only supports >= 5.6+. Sequel Pro supports older versions of MySQL, but doesn't support MySQL 8 (unlike Ace).

**Reply by stychos** [reply]
Right, just tested - looks like even sandboxed apps operate with the custom hosts file properly. Trouble not in it.

**Reply by bhjohns** [reply]
I believe the databases I'm connecting to are 5.6.  My local box is currently `mysql  Ver 15.1 Distrib 10.5.8-MariaDB, for osx10.15 (x86_64) using readline 5.1`

Everything should be 5.6 and above so I don't see why Ace wouldn't work and Sequel Pro would (plus terminal).  I assume there's something that's missing just can't figure out what.

**Reply by Jason-Morcos** [reply]
Going back to the core error message, the error is an inability to authenticate with MySQL. My hunch is either the MySQL version is older than 5.6, or you're using a connection plugin which Ace doesn't support. If you look for logs on your MySQL server, you may be able to get more info about why the server is rejecting the connection.

**Reply by bhjohns** [reply]
Unfortunately I don't have access to the database logs at this time since I'm not an admin.  I do think you might be on the right track though and it has something to do with the connector.  

Through some testing with some other connection UI's we got a different error message stating CLIENT_PLUGIN_AUTH is required and for one of the other UI's if we switched it from default to use the MySQL 5 Driver it worked.  Not sure if there's any way in Sequel Ace to select a different connector or expose more details from the error message?  I assume that Sequel Pro must be using an older or different connector

**Reply by Jason-Morcos** [reply]
Yeah, Sequel Ace is using a much newer MySQL client than Sequel Pro. Still sounding to me like an unsupported version of MySQL. Would love to see the result of the query `SHOW VARIABLES LIKE "%version%";`

**Reply by bhjohns** [reply]
I checked on both the database that works to connect to and the one that throws the error and they both have the exact same values:
Variable_name | Value
------------ | -------------
innodb_version | 5.6.23-72.1
protocol_version | 10
slave_type_conversions |
version | 5.6.23-72.1-log
version_comment | Percona Server (GPL), Release 72.1, Revision 0503478
version_compile_machine | x86_64
version_compile_os | Linux

What's weird is if it's the version of MySQL, why would it work for one of the databases and not the other?
